### PR TITLE
Support listing in res fs

### PR DESCRIPTION
**Pull request by fjmacagno**
Allow ls operations to be run on res filesystem objects, which is required for new tests in the deployer project.

### CODE REVIEW

**Review by changliang1007 — APPROVED** [review]
code changes look great to me!

Shall we update the https://github.com/Factual/uio/blob/master/README.md accordingly? users might think `res` still doesn't support `ls` based on the current doc

File: test/uio/fs/test_res.clj [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+(facts "Listing"
+  (fact "listing root succeeds"
+    (ls "res:///") =not=> [])
+  (fact "listing test dir gives correct results"
+    (count (ls "res:///test/")) => 1
+    (first (ls "res:///test")) => (has-suffix "test/test.txt")
+    (first (ls "res:///test/")) => (has-suffix "test/test.txt"))
+  (fact "listing test file returns just the same file"
+    (count (ls "res:///test/test.txt")) => 1
+    (first (ls "res:///test/test.txt")) => (has-suffix "test/test.txt")))
```
Comment (changliang1007) [inline test/uio/fs/test_res.clj]: btw does this behavior align with the `ls` on other file systems?

Comment (fjmacagno) [reply test/uio/fs/test_res.clj]: I believe so, at least now that i fixed it to be returning a map instead.

**Comment by fjmacagno** [comment]
Updated the README.

**Pull request merged**